include/soci/soci-platform.h

```diff
@@ -124,6 +124,119 @@ namespace cxx_details
     using std::auto_ptr;
 #endif
 
+#if defined(SOCI_HAVE_CXX11) || (defined(_MSC_VER) && _MSC_VER >= 1800)
```

Comment:
I think we should just start using C++11 (or, better, C++14) in SOCI master, see [this thread](https://sourceforge.net/p/soci/mailman/soci-devel/thread/E1lYsSp-0000NS-0C%40smtp.tt-solutions.com/#msg37265445).

OTOH I don't know if we need MT-safety provided by `std::shared_ptr` here, perhaps we should always use our simpler (and potentially more efficient) class instead?

At any rate, it would be better to have just version of it.

Reply:
Yes, please!

Since I am new around here and not a core maintainer, I would even talk about C++17. Since I am young, passionate about the new, and adventurous!

But very happy with C++11.